Projects/Server/Client/ArtData.cs

```diff
@@ -0,0 +1,185 @@
+        }
+
+        Span<ushort> buffer = stackalloc ushort[entry.Size / 2];
+        _dataStream.Seek(entry.Offset, SeekOrigin.Begin);
+        _dataStream.Read(MemoryMarshal.AsBytes(buffer));
+
+        var width = buffer[2];
+        var height = buffer[3];
+
+        if (width == 0 || height == 0)
+        {
+            return Rectangle2D.Empty;
+        }
+
+        return GetBoundsFromRGBA1555Bitmap(width, height, buffer[4..]);
+    }
+
+    private static Dictionary<int, UOPEntry> LoadMulRanges(string idxPath)
+    {
+        using var fs = new FileStream(idxPath, FileMode.Open, FileAccess.Read, FileShare.Read);
+        using var br = new BinaryReader(fs);
+
+        var count = (int)(fs.Length / 12);
+
+        var ranges = new Dictionary<int, UOPEntry>(count);
+
+        for (var i = 0; i < count; i++)
+        {
+            var offset = br.ReadInt32();
+            var length = br.ReadInt32();
+            br.ReadInt32(); // Skip the Data field since we don't use it
+
+            if (length > 0)
+            {
+                ranges[i] = new UOPEntry(offset, length);
+            }
+        }
+
+        ranges.TrimExcess();
+        return ranges;
+    }
+
+    public static Rectangle2D GetBoundsFromRGBA1555Bitmap(int width, int height, ReadOnlySpan<ushort> data)
+    {
+        ReadOnlySpan<ushort> lookups = data[..height];
+        data = data[height..];
+
+        int xMin = width;
+        int yMin = height;
+        int xMax = -1;
+        int yMax = -1;
+
+        for (int y = 0; y < height; y++)
+        {
+            var i = lookups[y];
+            var x = 0;
+
+            while (true)
+            {
+                int startX = data[i++];
+                var pixelCount = data[i++];
+
+                if (startX + pixelCount == 0 || startX > width)
+                {
+                    break;
+                }
+
+                x += startX;
+                for (var end = x + pixelCount; x < end; x++)
+                {
+                    // Get the pixel value from the bitmap data
+                    ushort pixel = data[i++];
+
+                    // Check if the pixel is non-black (0 in 555 RGB is completely black)
+                    if (pixel == 0)
+                    {
+                        continue;
+                    }
+
+                    if (x < xMin)
+                    {
+                        xMin = x;
+                    }
+
+                    if (x > xMax)
+                    {
+                        xMax = x;
+                    }
+
+                    if (y < yMin)
+                    {
+                        yMin = y;
+                    }
+
+                    if (y > yMax)
+                    {
+                        yMax = y;
+                    }
+                }
+            }
+        }
+
+        return xMax switch
+        {
+            // If no non-black pixels were found, return an empty rectangle
+            -1 => Rectangle2D.Empty,
```

Comment:
Note that in RunUO/ServUO, this value is `[0, 0, 0xFFFF, 0xFFFF]`. Intentionally different here. if the entire pic is black, there are no bounds, so we can keep it as all 00's. This also allows us to easily detect "empty" or "missing" entries without a validation structure.